ratis-grpc/src/main/java/org/apache/ratis/grpc/server/GrpcLogAppender.java

```diff
@@ -280,6 +271,16 @@ private void appendLog(boolean excludeLogEntries) throws IOException {
       }
     }
 
+    final long waitMs = getMinWaitTimeMs();
+    if (waitMs > 0) {
+      try {
+        Thread.sleep(waitMs);
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+        throw IOUtils.toInterruptedIOException(
+            "Interrupted appendLog, heartbeat? " + heartbeat, e);
+      }
+    }
```

Comment:
Shall we check the condition and wait inside `mayWait`?

Reply:
It won't work since `mayWait` can be waked up by `notifyLogAppender()`.

Reply:
@szetszwo I see. Thanks a lot for the reply!